Applications/FileManager/DirectoryView.cpp

```diff
@@ -108,8 +110,32 @@ void DirectoryView::set_view_mode(ViewMode mode)
     ASSERT_NOT_REACHED();
 }
 
+void print_path_history(int index, Vector<String> path_history)
+{
+    if (path_history.size() > 0) {
+        dbgprintf("File Manager: Path History:\n");
+        for (int i = 0; i < path_history.size(); i++) {
+            if (i == index)
+                dbgprintf("> ");
+            dbgprintf("Path: %s\n", path_history[i].view().characters());
+        }
+    }
+}
+
+void DirectoryView::add_path_to_history(const String& path)
+{
+    if (path_history_position < path_history.size())
```

Comment:
Vector::append does increase the capacity, so the resize here isn't necessary.

Small future note; perhaps it might be an idea to keep a cap on the size of the history? But no need to add that now, just a thought :)